### DISCUSSION TITLE
Trouble with ESP32-S3 and CameraWebServer

### DISCUSSION THREAD

**Discussion by istrian** [espressif/arduino-esp32#9249] [Q&A] [answered]
Hello,
Im working on my school project and i keep running into a problem with CameraWebServer and ESP32-S3. Im using ESP32-S3 N16R8 Development Board (https://www.aliexpress.com/item/1005006095863180.html?spm=a2g0o.order_list.order_list_main.35.6b7b1802oLfUvJ#nav-specification) with OV2640 camera.
I want to run the CameraWebServer code from arduino IDE.
**This is how i have configured the board:**
![image](https://github.com/espressif/arduino-esp32/assets/59991195/f21996ba-3895-4cd3-a4b9-876f050a6353)
**Here is the code:**
`#include "esp_camera.h"
#include <WiFi.h>

//
// WARNING!!! PSRAM IC required for UXGA resolution and high JPEG quality
//            Ensure ESP32 Wrover Module or other board with PSRAM is selected
//            Partial images will be transmitted if image exceeds buffer size
//
//            You must select partition scheme from the board menu that has at least 3MB APP space.
//            Face Recognition is DISABLED for ESP32 and ESP32-S2, because it takes up from 15 
//            seconds to process single frame. Face Detection is ENABLED if PSRAM is enabled as well

// ===================
// Select camera model
// ===================
//#define CAMERA_MODEL_WROVER_KIT // Has PSRAM
//#define CAMERA_MODEL_ESP_EYE // Has PSRAM
//#define CAMERA_MODEL_ESP32S3_EYE // Has PSRAM
//#define CAMERA_MODEL_M5STACK_PSRAM // Has PSRAM
//#define CAMERA_MODEL_M5STACK_V2_PSRAM // M5Camera version B Has PSRAM
//#define CAMERA_MODEL_M5STACK_WIDE // Has PSRAM
//#define CAMERA_MODEL_M5STACK_ESP32CAM // No PSRAM
//#define CAMERA_MODEL_M5STACK_UNITCAM // No PSRAM
//#define CAMERA_MODEL_AI_THINKER // Has PSRAM
//#define CAMERA_MODEL_TTGO_T_JOURNAL // No PSRAM
#define CAMERA_MODEL_XIAO_ESP32S3 // Has PSRAM
// ** Espressif Internal Boards **
//#define CAMERA_MODEL_ESP32_CAM_BOARD
//#define CAMERA_MODEL_ESP32S2_CAM_BOARD
//#define CAMERA_MODEL_ESP32S3_CAM_LCD
//#define CAMERA_MODEL_DFRobot_FireBeetle2_ESP32S3 // Has PSRAM
//#define CAMERA_MODEL_DFRobot_Romeo_ESP32S3 // Has PSRAM
#include "camera_pins.h"

// ===========================
// Enter your WiFi credentials
// ===========================
const char* ssid = "Gas";
const char* password = "gasermali";

void startCameraServer();
void setupLedFlash(int pin);

void setup() {
  Serial.begin(115200);
  Serial.setDebugOutput(true);
  Serial.println();

  camera_config_t config;
  config.ledc_channel = LEDC_CHANNEL_0;
  config.ledc_timer = LEDC_TIMER_0;
  config.pin_d0 = Y2_GPIO_NUM;
  config.pin_d1 = Y3_GPIO_NUM;
  config.pin_d2 = Y4_GPIO_NUM;
  config.pin_d3 = Y5_GPIO_NUM;
  config.pin_d4 = Y6_GPIO_NUM;
  config.pin_d5 = Y7_GPIO_NUM;
  config.pin_d6 = Y8_GPIO_NUM;
  config.pin_d7 = Y9_GPIO_NUM;
  config.pin_xclk = XCLK_GPIO_NUM;
  config.pin_pclk = PCLK_GPIO_NUM;
  config.pin_vsync = VSYNC_GPIO_NUM;
  config.pin_href = HREF_GPIO_NUM;
  config.pin_sccb_sda = SIOD_GPIO_NUM;
  config.pin_sccb_scl = SIOC_GPIO_NUM;
  config.pin_pwdn = PWDN_GPIO_NUM;
  config.pin_reset = RESET_GPIO_NUM;
  config.xclk_freq_hz = 20000000;
  config.frame_size = FRAMESIZE_UXGA;
  config.pixel_format = PIXFORMAT_JPEG; // for streaming
  //config.pixel_format = PIXFORMAT_RGB565; // for face detection/recognition
  config.grab_mode = CAMERA_GRAB_WHEN_EMPTY;
  config.fb_location = CAMERA_FB_IN_PSRAM;
  config.jpeg_quality = 12;
  config.fb_count = 1;
  
  // if PSRAM IC present, init with UXGA resolution and higher JPEG quality
  //                      for larger pre-allocated frame buffer.
  if(config.pixel_format == PIXFORMAT_JPEG){
    if(psramFound()){
      config.jpeg_quality = 10;
      config.fb_count = 2;
      config.grab_mode = CAMERA_GRAB_LATEST;
    } else {
      // Limit the frame size when PSRAM is not available
      config.frame_size = FRAMESIZE_SVGA;
      config.fb_location = CAMERA_FB_IN_DRAM;
    }
  } else {
    // Best option for face detection/recognition
    config.frame_size = FRAMESIZE_240X240;
#if CONFIG_IDF_TARGET_ESP32S3
    config.fb_count = 2;
#endif
  }

#if defined(CAMERA_MODEL_ESP_EYE)
  pinMode(13, INPUT_PULLUP);
  pinMode(14, INPUT_PULLUP);
#endif

  // camera init
  esp_err_t err = esp_camera_init(&config);
  if (err != ESP_OK) {
    Serial.printf("Camera init failed with error 0x%x", err);
    return;
  }

  sensor_t * s = esp_camera_sensor_get();
  // initial sensors are flipped vertically and colors are a bit saturated
  if (s->id.PID == OV3660_PID) {
    s->set_vflip(s, 1); // flip it back
    s->set_brightness(s, 1); // up the brightness just a bit
    s->set_saturation(s, -2); // lower the saturation
  }
  // drop down frame size for higher initial frame rate
  if(config.pixel_format == PIXFORMAT_JPEG){
    s->set_framesize(s, FRAMESIZE_QVGA);
  }

#if defined(CAMERA_MODEL_M5STACK_WIDE) || defined(CAMERA_MODEL_M5STACK_ESP32CAM)
  s->set_vflip(s, 1);
  s->set_hmirror(s, 1);
#endif

#if defined(CAMERA_MODEL_ESP32S3_EYE)
  s->set_vflip(s, 1);
#endif

// Setup LED FLash if LED pin is defined in camera_pins.h
#if defined(LED_GPIO_NUM)
  setupLedFlash(LED_GPIO_NUM);
#endif

  WiFi.begin(ssid, password);
  WiFi.setSleep(false);

  while (WiFi.status() != WL_CONNECTED) {
    delay(500);
    Serial.print(".");
  }
  Serial.println("");
  Serial.println("WiFi connected");

  startCameraServer();

  Serial.print("Camera Ready! Use 'http://");
  Serial.print(WiFi.localIP());
  Serial.println("' to connect");
}

void loop() {
  // Do nothing. Everything is done in another task by the web server
  delay(10000);
}`
**The error that Im getting from serial monitor:**
![image](https://github.com/espressif/arduino-esp32/assets/59991195/3733fba1-9aa8-45be-a086-ddd31e962911)
**I have tried:**
I have ESP-CAM, and Im able to get the code uploaded and the CameraWebServer working with the same OV2640 camera from ESP32-S3 but the problem is i cant get face recognition running as it has ESP32 inside it and it doesent support face recognition with it (that is why i switched to ESP32-S3)
Switching the USB port on my PC

Thank you for your help in advance.

**Comment by me-no-dev** [comment, marked as answer]
the error suggests that the driver can not find the sensor. That is usually due to bad pin config. make sure that your sccb pins are correct

**Reply by istrian** [reply]
Yea, thank you, that is true as when i select different board model it works, do you know where can i find the exact pin config for my board as i cant find the data sheet for it?

**Reply by Jason2866** [reply]
In the schematic diagram the are listed.
![IMG_0102](https://github.com/espressif/arduino-esp32/assets/24528715/ccad5773-09c4-4d1a-b6c4-1d2a6106341b)

**Reply by istrian** [reply]
Yes, thank you, i replaced those pins with the right ones and it works, for anyone who might have the same problem here they are:
![image](https://github.com/espressif/arduino-esp32/assets/59991195/9e54e860-8d6c-4192-b0ec-8f24579819eb)

**Reply by Utopists** [reply]
I just want to thank you for this post and the solution.

**Reply by Recycling4p0** [reply]
@istrian Hi,
I know it has been few months since the answer to this post but I was wondering if your code still works? I said this because I too has recently bought the same exact ESP32-S3 module on aliexpress and has configured everything including your camera pins and the same exact code as here but I still get the

E (571) esp_core_dump_flash: No core dump partition found!

Could this be related to the newest version of Arduino? Alternatively the board is seen by default as: 4d systems gen4-esp32 16MB Modules (ESP32-S3 r8n16) but I have changed to ESP32S3 Dev Board for the sake of trying out your code since you mentioned it worked last time. Any Clue?

Thanks heaps in advance

**Comment by istrian** [comment]
Hey,
I cant upload the code as i have wiped my pc and no longer have access to
it, please try viewing the forum:
https://www.reddit.com/r/esp32/comments/1797tr4/esp_core_dump_flash_no_core_dump_partition_found/

**Comment by Recycling4p0** [comment]
Hi istrian, thanks heaps for getting back to my question. Really prob the most important reason I asked was because I was using the same module as you bought from AliExpress and it’s an N16R8 based whereas the link you suggested above is about the Wrover modules. Did you really get the web server to run? Because I have tested 4 modules already and while they work for light applications they all fail to use PSRAM when I attempt to use camera, the is also a malloc buffer error along with PSRAM ID check error is 0x0000000,

**Comment by Recycling4p0** [comment]
It seems the external 8MB PSRAM has to do with the error! It has been over 2 weeks and I would really like if someone can shed some light! Thanks in advance

**Comment by istrian** [comment]
Try making a new thread if you havent already so more people can see your
issue.
